Feature/#505 add quiz fsm (#520)
CircuitVerse/Interactive-Book: docs/seq-design/fsm.md

{% include fsm.html %}

Lets try the soda can vending machine and look for the corresponding state changes FSM diagram in each of the above mentioned possibilities.

{:.quiz}

1. Behavior of synchronous sequential circuits represented in the graphical form is called `_____`.
   1. State diagram
   * Block diagram
   * Flow diagram
   * All of these
2. Which of the following is a type of `FSM(Finite State Machine)`?
   1. Mealy State Machine
   * Moore State Machine
   * Both 
   * None   

> 3. Finite State Machine is said to be `Mealy state machine`?

Reviewer: ```suggestion
3. In `Mealy state machine` the outputs depends on ?
```

>    1. Outputs depend on present inputs & present states.

Reviewer: ```suggestion
   1. on present inputs & present states.
```

>    * Outputs depend only on present states.  

Reviewer: ```suggestion
   * Only on present states.  
```

> 3. Finite State Machine is said to be `Moore state machine`?

Reviewer: make same changes Here also
